reducer/src/test/java/com/graphicsfuzz/reducer/reductionopportunities/ExprToConstantReductionOpportunitiesTest.java

```diff
@@ -98,4 +98,83 @@ public void testSwitch() throws Exception {
     assertEquals(0, ops.size());
   }
 
+  @Test
+  public void testSimplifyLiterals() throws Exception {
+    final String program = "#version 310 es\n"
+        + "struct S {\n"
+        + "  int a;\n"
+        + "  float b;\n"
+        + "};\n"
+        + "void main() {\n"
+        + "  100;\n"
+        + "  200.0;\n"
+        + "  300u;\n"
```

Comment:
What if you have a switch with a few different int cases? Will they all get simplified to the same thing?

Reply:
This was already catered for in ReductionOpportunitiesBase, via:

```
  @Override
  public void visitExprCaseLabel(ExprCaseLabel exprCaseLabel) {
    // Do *not* invoke super.visitExprCaseLabel(...), as we do not wish to look for opportunities
    // to simplify the literal expression that is used as a case label.  Literals are simple enough
    // already, and attempting to change the value of a case label literal runs the risk of
    // introducing duplicate case labels.
  }
```

However, I have added an extra test for it just to be sure.